joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/passes/AstCreator.scala

```diff
@@ -726,7 +726,11 @@ class AstCreator(filename: String, cls: SootClass, global: Global)(implicit with
             .argumentIndex(0)
             .dynamicTypeHintFullName(Seq(parentType))
         case _: NewMethodParameterIn =>
-          NodeBuilders.newThisParameterNode(parentType, Seq(parentType), line(Try(method.tryResolve()).getOrElse(null)))
+          NodeBuilders.newThisParameterNode(
+            typeFullName = parentType,
+            dynamicTypeHintFullName = Seq(parentType),
+            line = line(Try(method.tryResolve()).getOrElse(null))
```

Comment:
Is there not a way to have this `line` method accept `Option` types? Could go `Try(method.tryResolve()).toOption`

Reply:
`line(Try(method.tryResolve()).getOrElse(null))` method which is getting called returns and `Option` check line no `1148`

Reply:
Ok could we try
```scala
line = Try(method.tryResolve()).toOption.map(line)
```

Reply:
line() expects a `Host` object in this map it passes `Option`. Hence it throws error.

I have just updated the method call to use named parameters with the change I made in the method parameters.

![image](https://github.com/joernio/joern/assets/5101898/422484b9-5ed3-4c63-b2aa-39c48facfede)